content/blog/scheduling-in-react/index.md

```diff
@@ -0,0 +1,237 @@
+
+<!-- prettier-ignore -->
+```js
+function App() {
+  const [searchValue, setSearchValue] = React.useState();
+
+  function handleChange(value) {
+    setSearchValue(value);
+  }
+
+  return (
+    <div>
+      <SearchBox onChange={handleChange} />
+      <NameList searchValue={searchValue} />
+    </div>
+  );
+}
+
+function SearchBox(props) {
+  const [inputValue, setInputValue] = React.useState();
+
+  function handleChange(event) {
+    const value = event.target.value;
+
+    setInputValue(value);
+    props.onChange(value);
+    sendAnalyticsPing(value);
+  };
+
+  return (
+    <input
+      type="text"
+      value={inputValue}
+      onChange={handleChange}
+    />
+  );
+}
+
+ReactDOM.render(<App />, container);
+```
+
+_ℹ️ This example uses [React Hooks](https://reactjs.org/docs/hooks-intro.html). If you're not familiar with them yet, take a look at the [CodeSandbox code](https://codesandbox.io/s/j3zrqpzkr5). Addition you might wonder why we use two different state variables for this example. We'll find out why in a bit._
+
+The `<SearchBox>` displays the input element which is controlled by the local state variable `inputValue`. When the input changes, it will call the `onChange` prop with the new value. The new value is then stored inside the `<App>` component's `searchValue` state and passed over to the `<NameList>`. At the end, we also simulate an analytics ping to our servers.
+
+Try it out! Type a name (e.g. "Adeline") in the search box below and see how it works:
+
+<iframe src="https://codesandbox.io/embed/j3zrqpzkr5?fontsize=14&hidenavigation=1&view=preview" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
+
+You can notice that the interface is not very responsive. To highlight the issue, I artificially slowed down the rendering time of the names list. And since this list is big, it has a significant impact on the applications performance. This is not good 😰.
+
+Our users expect immediate feedback but the app is unresponsive for seconds after a keystroke. To understand what's going on, let's take a look at the DevTools' Performance tab. Here's a screenshot of a recording while I type the word "Ada" into the search box:
+
+![Screenshot of Chrome DevTools that show that the three keypress events takes 733ms to render](devtools-sync.png)
+
+We can see that there are a lot of red triangles which is usually not a good sign. For every keystroke, we see a `keypress` event being fired. All of the three events are run within one frame[^4] causing that frame to take **733ms**. That's way above our average frame budget of 16ms.
```

Comment:
"All of the three events are run within one frame"
I think you can remove the "are"